### DISCUSSION TITLE
The type or namespace name 'App' could not be found (are you missing a using directive or an assembly reference?)

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#15260] [Q&A] [answered]
Trying to compile my program on OSX ARM (Apple M1) and it fails in program.cs

```
using Avalonia;
using System;

namespace SelectPicturesAvalonia.OSX
{
    internal sealed class Program
    {
        // Initialization code. Don't use any Avalonia, third-party APIs or any
        // SynchronizationContext-reliant code before AppMain is called: things aren't initialized
        // yet and stuff might break.
        [STAThread]
        public static void Main (string [] args) => BuildAvaloniaApp ()
            .StartWithClassicDesktopLifetime (args);

        // Avalonia configuration, don't remove; also used by visual designer.
        public static AppBuilder BuildAvaloniaApp ()
            => AppBuilder.Configure<App> ()                               <<<< Here
                .UsePlatformDetect ()
                .WithInterFont ()
                .LogToTrace ();
    }
}

```
on line 16 with the message in the title.

The program runs on Windows, Android, IOS emulator (my IPad is too old), but NOT on Mac.
What could be wrong here ?

Apple M1 (8GB - Sonoma 14.4.1 - 256GB HD)
Visual Studio for Mac 17.6.10 (build 428)
.NET 8.0.203

**Comment by thevortexcloud** [comment, marked as answer]
Is your App class included in the compilation output on Mac? It might not be if it's in a platform specific project.

**Reply by Yveske1966** [reply]
The Microsoft.NETCore.App is listed under Dependencies - Frameworks if that is what you are referring to.
App.xaml and App.xaml.cs are part of the "standard" program and the .Android, .Desktop, .IOS and .OSX only contain platform specific files like program.cs, 

<img width="1920" alt="Screenshot 2024-04-07 at 06 39 51" src="https://github.com/AvaloniaUI/Avalonia/assets/68047793/c079d716-e025-433c-b86e-2f22924f8484">

**Reply by Yveske1966** [reply]
Found the documentation to build Avalonia on Mac (https://github.com/AvaloniaUI/avalonia-docs/blob/main/docs/guides/platforms/macos-development.md)
Found out how to compile the headers first (https://github.com/AvaloniaUI/Avalonia/issues/4991)
Compiling Avalonia reported a lot of warnings (mostly deprecated this and that), but it gave my a dylib
Added .with to the program.cs but still no go...

using Avalonia;
using System;

namespace SelectPicturesAvalonia.OSX
{
    internal sealed class Program
    {
        // Initialization code. Don't use any Avalonia, third-party APIs or any
        // SynchronizationContext-reliant code before AppMain is called: things aren't initialized
        // yet and stuff might break.
        [STAThread]
        public static void Main (string [] args) => BuildAvaloniaApp ()
            .StartWithClassicDesktopLifetime (args);

        // Avalonia configuration, don't remove; also used by visual designer.
        public static AppBuilder BuildAvaloniaApp ()
            => AppBuilder.Configure<App> ()
                .UsePlatformDetect ()
                .WithInterFont ()
                .LogToTrace ()
                .With (new AvaloniaNativePlatformOptions
                {
                    AvaloniaNativeLibraryPath = "/Users/yvesdepreitere/Repos/SelectPicturesAvalonia/SelectPicturesAvalonia/SelectPicturesAvalonia.OSX/libAvalonia.Native.OSX.dylib",
                });
    }
}

Does anyone know how to proceed ?

**Reply by Yveske1966** [reply]
No need to change program.cs
